flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/SlotSharingSlotAllocator.java

```diff
@@ -148,7 +149,17 @@ public ResourceCounter calculateRequiredSlots(
 
     @Override
     public Map<ExecutionVertexID, LogicalSlot> reserveResources(
-            VertexParallelismWithSlotSharing vertexParallelismWithSlotSharing) {
+            VertexParallelism vertexParallelism) {
+        Preconditions.checkArgument(
+                vertexParallelism instanceof VertexParallelismWithSlotSharing,
```

Comment:
True, this should also work. Given that this is an internal API and that changing it would require some extra work, I would keep it as is and maybe change it in the future if we want.